libraries/libbase/source/org/pentaho/reporting/libraries/base/util/XMLParserFactoryProducer.java

```diff
@@ -0,0 +1,42 @@
+/*!
+* This program is free software; you can redistribute it and/or modify it under the
+* terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+* Foundation.
+*
+* You should have received a copy of the GNU Lesser General Public License along with this
+* program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+* or from the Free Software Foundation, Inc.,
+* 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+* without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+* See the GNU Lesser General Public License for more details.
+*
+* Copyright (c) 2002-2016 Pentaho Corporation..  All rights reserved.
+*/
+
+package org.pentaho.reporting.libraries.base.util;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+public class XMLParserFactoryProducer {
+
+  /**
+   * Creates an instance of {@link DocumentBuilderFactory} class with enabled {@link XMLConstants#FEATURE_SECURE_PROCESSING} property.
+   * Enabling this feature prevents from some XXE attacks (e.g. XML bomb)
+   * See PPP-3506 for more details.
+   *
+   * @throws ParserConfigurationException if feature can't be enabled
+   *
+   */
+  public static DocumentBuilderFactory createSecureDocBuilderFactory() throws ParserConfigurationException {
+    DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+    docBuilderFactory.setFeature( XMLConstants.FEATURE_SECURE_PROCESSING, true );
+    docBuilderFactory.setFeature( "http://apache.org/xml/features/disallow-doctype-decl", true );
```

Comment:
I would argue that this should be wrapped in a try-catch. If you choose a non-apache based XML parser, you should not end up with an unrecoverable error. Log a warning, and then let the user parse his XML file.